test/e2e/ctrcfg_test.go

```diff
@@ -0,0 +1,163 @@
+package e2e_test
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+	"time"
+
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/openshift/machine-config-operator/test/e2e/framework"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func TestContainerRuntimeConfigPidsLimit(t *testing.T) {
+	runTestWithCtrcfg(t, "pids-limit", `pids_limit = (\S+)`, "12345", &mcfgv1.ContainerRuntimeConfiguration{
+		PidsLimit: 12345,
+	})
+}
+
+// runTestWithCtrcfg creates a ctrcfg and checks whether the expected updates were applied, then deletes the ctrcfg and makes
+// sure the node rolled back as expected
+// testName is a string to identify the objects created (MCP, MC, ctrcfg)
+// regex key is the searching critera in the crio.conf. It is expected that a single field is in a capture group, and this field
+//   should equal expectedConfValue upon update
+// cfg is the ctrcfg config to update to and rollback from
+func runTestWithCtrcfg(t *testing.T, testName, regexKey, expectedConfValue string, cfg *mcfgv1.ContainerRuntimeConfiguration) {
+	cs := framework.NewClientSet("")
+	matchValue := fmt.Sprintf("%s-%s", testName, uuid.NewUUID())
+	ctrcfgName := fmt.Sprintf("ctrcfg-%s", matchValue)
+	poolName := fmt.Sprintf("node-%s", matchValue)
+	mcName := fmt.Sprintf("mc-%s", matchValue)
+
+	// label one node from the pool to specify which worker to update
+	unlabelFunc := labelRandomNodeFromPool(t, cs, "worker", mcpNameToRole(poolName))
+	defer unlabelFunc()
```

Comment:
I think I understand what's going on here, follow https://github.com/openshift/machine-config-operator/blob/master/docs/custom-pools.md#removing-a-custom-pool

Basically, first thing unlabel the node, in this function you should wait for the pool to go idle as it's going back to worker - then only after go ahead and cleanup 1) MC 2) MCP

Reply:
rtfm @haircommander :roll_eyes: 

thanks!